src/components/molecules/nav/navigation.tsx

```diff
@@ -1,51 +1,48 @@
-/* eslint-disable */
-// This file was converted to .tsx without actually implementing typescript
-// @TODO: update this file to tsx and enable eslint
-import React, { Component } from 'react'
+import React, { Component, ReactNode } from 'react'
 import { Link } from 'gatsby'
 
 import styles from './navigation.module.css'
 
-import Menu from '../../../img/menu.inline.svg'
-import Close from '../../../img/close.inline.svg'
-import Down from '../../../img/down.inline.svg'
+import { Menu } from '../../../img/menu.inline'
+import { Close } from '../../../img/close.inline'
+// import Down from '../../../img/down.inline.svg'
 
 import classNames from 'classnames/bind'
 const cx = classNames.bind(styles)
 
-export default class Nav extends Component {
-  state = {
+type NavProps = {
+  light: boolean
+  navlink?: boolean
+}
+
+type State = {
+  revealed: boolean
+  nextShown: boolean
+}
+
+export default class Nav extends Component<NavProps, State> {
+  readonly state = {
     revealed: false,
     nextShown: false,
   }
 
-  reveal = (e) => {
-    e.preventDefault()
+  reveal: React.MouseEventHandler<HTMLSpanElement> = () => {
```

Comment:
Interesting, never seen this before

Reply:
Me either, but I needed to do it in order to make typescript happy :)